test/test_array.py

```diff
@@ -309,9 +310,11 @@ def test_array_context_einsum_array_tripleprod(actx_factory, spec):
 
 # {{{ test array container
 
+@with_container_arithmetic(bcast_obj_array=True, rel_comparison=True)
 @dataclass_array_container
+@has_array_context_attribute
 @dataclass(frozen=True)
```

Comment:
Four class decorators? Kind of ridiculous, especially given that this week is the first time I've ever written one, but... I don't know an alternative that I'd prefer.

Reply:
Just provide some wrappers for "dataclass with arithmetic" and some other common cases as they show up?

Reply:
Not sure I'd prefer that... at least this way is fairly composable.

Reply:
I meant having both this and some helpers. Those are fairly easy to add if these building blocks are there, so I don't feel strongly about it.

Reply:
Fair. We just lost `has_array_context_attribute` though, which makes this less urgent IMO.